File: python/223_Rectangle Area.py
#!/usr/bin/env python2

class Solution(object):
    def computeArea(self, A, B, C, D, E, F, G, H):
        """
        :type A: int
        :type B: int
        :type C: int
        :type D: int
        :type E: int
        :type F: int
        :type G: int
        :type H: int
        :rtype: int
        """

        # 没有相交的情况
        if C <= E or G <= A or H <= B or D <= F:
            return (C - A) * (D - B) + (G - E) * (H - F)

        # 相交的情况
        left = max(A, E)
        right = min(C, G)
        top = min(D, H)
        bottom = max(B, F)

        return (C - A) * (D - B) + (G - E) * (H - F) - (top - bottom) * (right - left)

File: java/src/brand_new/_223_Rectangle_Area.java
package brand_new;

public class _223_Rectangle_Area {
	public int computeArea(int A, int B, int C, int D, int E, int F, int G,
			int H) {
		return (int) ((C - A) * (D - B) + (G - E) * (H - F) - ((long) Math.min(
				C, G) - Math.max(A, E) > 0 ? ((long) Math.min(C, G) - Math.max(
				A, E)) : 0)
				* ((long) Math.min(D, H) - Math.max(B, F) > 0 ? ((long) Math
						.min(D, H) - Math.max(B, F)) : 0));
	}

	public static void main(String[] args) {
		System.out.println(new _223_Rectangle_Area().computeArea(-1500000001,
				0, -1500000000, 1, 1500000000, 0, 1500000001, 1));
	}

}
